### PR TITLE
Current version of GCToo code 

### PR DESCRIPTION
**Pull request by oena**

### CODE REVIEW

File: python/broadinstitute/cmap/io/GCToo/parse_gctoox.py [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+		- rid (list of strings): list of row ids to specifically keep from gctx. Default=None. 
+		- cid (list of strings): list of col ids to specifically keep from gctx. Default=None. 
+
+	Output:
+		- df: subsetted DataFrame. 
+	"""
+	if rid == None and cid == None:
+		pass
+	elif rid != None and cid != None:
+		df = df.loc[rid][cid]
```
Comment (levlitichev) [inline python/broadinstitute/cmap/io/GCToo/parse_gctoox.py]: The notation I'm familiar with is df.loc[rid, cid]. It looks like this works too!

File: python/broadinstitute/cmap/io/GCToo/test_write_gctoox.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+		self.assertTrue(set(list(mini_gctoo.col_metadata_df.columns)) == set(list(mini_gctoo_col_metadata.columns)),
+			"Mismatch between expected col metadata columns {} and column values written to file: {}".format(mini_gctoo.col_metadata_df.columns, mini_gctoo_col_metadata.columns))
+		self.assertTrue(set(list(mini_gctoo.col_metadata_df.index)) == set(list(mini_gctoo.col_metadata_df.index)),
+			"Mismatch between expect col metadata index {} and index values written to file: {}".format(mini_gctoo.col_metadata_df.index, mini_gctoo_col_metadata.index))
+		for c in list(mini_gctoo.col_metadata_df.columns):
+			self.assertTrue(set(list(mini_gctoo.col_metadata_df[c])) == set(list(mini_gctoo_col_metadata[c])), 
+				"Values in column {} differ between expected metadata and written col metadata!".format(c))
+
+	# TODO: add test case with id metadata 
+	# Also look into R adding/not adding an id field
```
Comment (levlitichev) [inline python/broadinstitute/cmap/io/GCToo/test_write_gctoox.py]: @oena, it sounds like you might have already taken care of this.

File: python/broadinstitute/cmap/io/GCToo/test_GCToo.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+        out_gctoo = GCToo.slice(my_gctoo, row_bool, col_bool)
+
+        self.assertEqual(out_gctoo.data_df.shape, (3, 2),
+                         'Size of returned data_df is wrong: {}'.format(out_gctoo.data_df.shape))
+        self.assertTrue(np.array_equal(out_gctoo.data_df.index.values, np.array(['a','c','d'])),
+                         'Incorrect data rows returned: {}'.format(out_gctoo.data_df.index.values))
+        self.assertTrue(np.array_equal(out_gctoo.col_metadata_df.index.values, np.array(['e','f'])),
+                         'Incorrect column metadata rows returned: {}'.format(out_gctoo.col_metadata_df.index.values))
+
+        # TODO(lev): test edge case of only one row or sample remaining
```
Comment (levlitichev) [inline python/broadinstitute/cmap/io/GCToo/test_GCToo.py]: Slicing will soon be its own file, so this will be taken care of.

File: python/broadinstitute/cmap/io/GCToo/concat_gctoo.py [hunk 4]
```diff
@@ -0,0 +1,285 @@
+import argparse
```
Comment (levlitichev) [inline python/broadinstitute/cmap/io/GCToo/concat_gctoo.py]: I've got a new version of this script that I can add once we accept the pull request.

**Comment by levlitichev** [comment]
Things look good to me! Nice job, @oena.
